lib/src/types.rs

```diff
@@ -62,20 +62,20 @@ pub enum Stmt {
     TypedStmt(Box<Stmt>, Type),
 }
 
-#[derive(Debug, Clone, Deserialize, Serialize)]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
 pub enum StmtExpr {
     Assign(String, Expr),
     New(Type, Vec<Expr>),
     MethodCall(Expr, String, Vec<Expr>),
     TypedStmtExpr(Box<StmtExpr>, Type),
 }
 
-#[derive(Debug, Clone, Deserialize, Serialize)]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
 pub enum Expr {
     This,
-    Super,
     LocalOrFieldVar(String),
     InstVar(Box<Expr>, String),
+    LocalVar(Box<Expr>, String),
```

Comment:
What are the parameters to `LocalVar` supposed to mean? The String is probably the name of the variable, right? But why save another `Expr` with it?

Also, are you expecting the Parser to already use the `LocalVar` node in the AST or do you only want to add it during the typechecking phase (like the `TypedExpr` for example).

Also also why did you feel the need to split `LocalVar` from `LocalOrFieldVar` anyways? I'm sure there's a good reason for it, I just don't see it rn

Reply:
We need the `LocalVar` type (and actually a `FieldVar` type) to replace `LocalOrFieldVar` in the TAST. `LocalVar` in fact does not need a `Box<Expr>` as it only derives the name from `LocalOrFieldVar` in the the form of a string. We should remove that.

Reply:
Sounds like a good idea 👍